packages/datetime/src/common/dateUtils.ts

```diff
@@ -110,6 +110,27 @@ export function isMonthInRange(date: Date, dateRange: DateRange) {
     return start <= day && day <= end;
 }
 
+export function isTimeInRange(date: Date, minDate: Date, maxDate: Date): boolean {
+    const time = getDateOnlyWithTime(date).getTime();
+    const minTime = getDateOnlyWithTime(minDate).getTime();
+    const maxTime = getDateOnlyWithTime(maxDate).getTime();
+
+    const isTimeGreaterThanMinTime = time >= minTime;
+    const isTimeSmallerThanMaxTime = time <= maxTime;
+
+    if (maxTime <= minTime) {
```

Comment:
@giladgray can you offer your thoughts here if possible?

@JacekJagiello I definitely see where you're coming from. I've thought about this more: if we add a function-description comment above describing the handling of the overlapping case, I'd be fine going with the approach you've already implemented here.